zh/before-start_en.md

```diff
@@ -0,0 +1,62 @@
+# Before the beginning
+## The target audience for this document
+### The confused software developers
+* Developers who want to write code that can be easily read but don't know how to do that  
+* Developers who want to understand other people's code  
+### Software development beginner
+This is a document for software development beginners, who can be defined as having learned no more than two languages and being able to complete most of exercises in the course in terms of the language skills they have learned. From the general situation of college education, they could basicly complete the final course assignments arranged by teachers. We may not be able to say that they have done very well. It is also difficult to improve further.  

```

Comment:
The document is for software development beginners. The so-called beginners are the ones who learned no more than two programming languages, and for the acquired programming skills, they can complete most of exercises in the course. For general college education, they could basically complete the final homework assigned by teachers. We can't say that they do it very well, and it is difficult to get further development.